### PR TITLE
Editorial: Tweak Array exotic object internal methods

### PR DESCRIPTION
**Pull request by shvaikalesh**
This change replaces prose in normative steps with [`IsDataDescriptor`](https://tc39.es/ecma262/#sec-isdatadescriptor) abstract op, explicitly marks a `NOTE:` and result of `OrdinaryDefineOwnProperty` call.

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -8976,7 +8978,8 @@ <h1>ArraySetLength ( _A_, _Desc_ )</h1>
               1. Perform ! OrdinaryDefineOwnProperty(_A_, *"length"*, _newLenDesc_).
               1. Return *false*.
           1. If _newWritable_ is *false*, then
-            1. Return OrdinaryDefineOwnProperty(_A_, *"length"*, PropertyDescriptor { [[Writable]]: *false* }). This call will always return *true*.
+            1. Let _succeeded_ be ! OrdinaryDefineOwnProperty(_A_, *"length"*, PropertyDescriptor { [[Writable]]: *false* }).
+            1. Assert: _succeeded_ is *true*.
```
Comment (michaelficarra) [inline spec.html]: I don't know why we care to assert this. We can just call `OrdinaryDefineOwnProperty` for its effects and be done with it.

Comment (ljharb) [reply spec.html]: It returns `false` if it fails, so without the assertion, a failure would be silently allowed and ignored.

Comment (michaelficarra) [reply spec.html]: But it won't fail.

Comment (ljharb) [reply spec.html]: Sure - that's what the note was meant to convey, and what this assert conveys. It seems important to preserve the intent that this call should never be able to be made to fail in the future, as well.

Comment (shvaikalesh) [reply spec.html:8982]: The assert might be taking an extra step, but I find it helpful because descriptors merging logic in [`ValidateAndApplyPropertyDescriptor`](https://tc39.es/ecma262/#sec-validateandapplypropertydescriptor) is a bit hard to follow. There is another assert like this one in Array's [`[[DefineOwnProperty]]`](https://tc39.es/ecma262/#sec-array-exotic-objects-defineownproperty-p-desc).
